eng/common/native/init-distro-rid.sh

```diff
@@ -11,67 +11,66 @@
 #   non-portable rid
 getNonPortableDistroRid()
 {
-    local targetOs="$1"
-    local targetArch="$2"
-    local rootfsDir="$3"
-    local nonPortableRid=""
+    targetOs="$1"
+    targetArch="$2"
+    rootfsDir="$3"
+    nonPortableRid=""
 
     if [ "$targetOs" = "linux" ]; then
+        # shellcheck disable=SC1091
         if [ -e "${rootfsDir}/etc/os-release" ]; then
-            source "${rootfsDir}/etc/os-release"
-
-            if [[ "${ID}" == "rhel" || "${ID}" == "rocky" || "${ID}" == "alpine" ]]; then
-                # remove the last version digit
-                VERSION_ID="${VERSION_ID%.*}"
+            . "${rootfsDir}/etc/os-release"
+            if [ "${ID}" = "rhel" ] || [ "${ID}" = "rocky" ] || [ "${ID}" = "alpine" ]; then
+                VERSION_ID="${VERSION_ID%.*}" # Remove the last version digit for these distros
```

Comment:
We stopped using it this trimming type of expansion for __solaris_major_version but not here. Is there difference in shell supports?

Reply:
Ah, ChatGPT thinks this is the way to go, and shellcheck doesn't complain.
Bourne Shell:
![image](https://github.com/dotnet/runtime/assets/3840695/0580d14b-77af-49cc-a4fe-9681abdfc9ac)


But good eye! I'll make it consistent later today.

Reply:
ChatGPT's reply:
> If your target environments are systems that are 15 years old or newer, you are generally in a safe zone regarding the support for ${variable%pattern} parameter expansion. This feature is part of the POSIX standard, and most Unix-like systems from the last 15 years would have /bin/sh linked to a shell that supports POSIX parameter expansion

so advanced ops like substring extraction `${string:position:length}` are bash-specific. So I think we can safely use that parameter expansion syntax for now.